stdlib/sqlite3/dbapi2.pyi

```diff
@@ -312,27 +313,26 @@ else:
     def register_adapter(type: type[_T], caster: _Adapter[_T], /) -> None: ...
     def register_converter(name: str, converter: _Converter, /) -> None: ...
 
-class _AggregateProtocol(Protocol):
-    def step(self, value: int, /) -> object: ...
-    def finalize(self) -> int: ...
+class _SingleParamAggregateProtocol(Protocol[_SQLType]):
+    def step(self, param: _SQLType, /) -> object: ...
+    def finalize(self) -> _SQLType: ...
 
-class _SingleParamWindowAggregateClass(Protocol):
-    def step(self, param: Any, /) -> object: ...
-    def inverse(self, param: Any, /) -> object: ...
-    def value(self) -> _SqliteData: ...
-    def finalize(self) -> _SqliteData: ...
+class _AnyParamAggregateProtocol(Protocol[_SQLType]):
+    def step(self, *args: _SQLType) -> object: ...
+    def finalize(self) -> _SQLType: ...
 
-class _AnyParamWindowAggregateClass(Protocol):
-    def step(self, *args: Any) -> object: ...
-    def inverse(self, *args: Any) -> object: ...
-    def value(self) -> _SqliteData: ...
-    def finalize(self) -> _SqliteData: ...
+class _SingleParamWindowAggregateClass(Protocol[_SQLType]):
+    def step(self, param: _SQLType, /) -> object: ...
+    def inverse(self, param: _SQLType, /) -> object: ...
+    def value(self) -> _SQLType: ...
+    def finalize(self) -> _SQLType: ...
+
+class _AnyParamWindowAggregateClass(Protocol[_SQLType]):
+    def step(self, *args: _SQLType) -> object: ...
+    def inverse(self, *args: _SQLType) -> object: ...
+    def value(self) -> _SQLType: ...
+    def finalize(self) -> _SQLType: ...
 
-class _WindowAggregateClass(Protocol):
-    step: Callable[..., object]
```

Comment:
From testing things out, it doesn't seem this protocol really worked as intended in either MyPy or Pyright. Unless you actually were annotating a lambda perhaps. Due to this, I went ahead and removed it.

Some examples of how it might not work as you would expect:

[Pyright Playground
](https://pyright-play.net/?pythonVersion=3.13&strict=true&code=GYJw9gtgBALgngBwJYDsDmUkQWEMoDCAhgDYlEBGJApgDRQAK4MYAxmCfQIIpwBQfVuQDOwqAFk4TMC3YkAFNNkcAlAC4%2BULVAhwA%2Bq1Lkq1NYSOUaAbQB0d%2BmAoAraqxgBdTdq9bBIsZIE-hraUAAm1MA6%2BoZkltTywtQkwPREIGh6AIxmPHBpGXoATLm8KlAAtAB8UAByYCimPqFQINQwAK4gKHUN1M2hA94RUcBgYPIIzGBmkkpsqpU19Y0h2nY2zXxjE4H%2B8ioqQA)

[MyPy Playground](https://mypy-play.net/?mypy=latest&python=3.12&flags=strict&gist=a1b87e82dfb5620fd970c7625ec50fd6)

Reply:
This has been part of the initial commit in #7625, while the other protocols already used a function. Maybe @JelleZijlstra remembers why we used an attribute instead of a function here?

Reply:
I don't really remember what I was thinking when I wrote that code, but the annotations proposed in this PR mean that protocol implementations must take `*args`. I am not familiar with how these things are used, but I'd expect concrete implementations to only accept a fixed number of parameters. Maybe that's why I chose to use `Callable[...`.

Reply:
I tried messing around with TypeVarTuples to do that, but had some issues there as well, there might not be a great way, but I'll see if I can figure something out.